pkg/ocmagenthandler/ocmagenthandler_networkpolicy.go

```diff
@@ -0,0 +1,110 @@
+package ocmagenthandler
+
+import (
+	"context"
+	"reflect"
+
+	netv1 "k8s.io/api/networking/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	ocmagentv1alpha1 "github.com/openshift/ocm-agent-operator/pkg/apis/ocmagent/v1alpha1"
+	oah "github.com/openshift/ocm-agent-operator/pkg/consts/ocmagenthandler"
+)
+
+func buildNetworkPolicy() netv1.NetworkPolicy {
+	namespacedName := oah.BuildNamespacedName(oah.OCMAgentNetworkPolicyName)
+	np := netv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      namespacedName.Name,
+			Namespace: namespacedName.Namespace,
+		},
+		Spec: netv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": oah.OCMAgentName,
```

Comment:
do we want to restrict the access to the ocm-agent pods only or the whole project?

Reply:
It's possible, though we'd also need rules for whatever prometheus components crawl the metrics if we block off the whole namespace, gets a bit more complicated.

Reply:
Oh right, I get you. I'm not sure that is easily do-able in a NetworkPolicy rule though.
To apply the rules to the whole namespace it would mean leaving the `PodSelector` out.
But the result of doing that blocks everything except AlertManager from being able to communicate to pods in the namespace. It would mean that the ingress rules would need to define every expected ingress into that namespace, not just AlertManager.

Reply:
I think we just need to leave the `spec.podSelector` to blank here. And the existing `spec.Ingress` should be able to filter the alertmanger in openshift-monitoring ns.
Just I am not sure if we want all the pods be accessible from the alertmanager.

Like:
```
spec:
   podSelector: {}
   ingress:
   - from: 
     - podSelector:
           matchLabels:
               app: alertmanager
     - namespaceSelector:
           matchLabels:
                name: openshift-monitoring
```